### PR TITLE
Increase GHO Facilitator Capacity

### PR DESCRIPTION
**Pull request by efecarranza**
# Changelog

Increase GHO facilitator capacity.

### CODE REVIEW

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.sol [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+/**
+ * @title Increase CCIP Facilitator Capacity
+ * @author @karpatkey_TokenLogic
+ * - Snapshot: Direct-to-AIP
+ * - Discussion: https://governance.aave.com/t/arfc-gho-arb-increase-ccip-facilitator-capacity/18169
+ */
+contract AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707 is IProposalGenericExecutor {
+  address public constant GHO_TOKEN_POOL = 0x5756880B6a1EAba0175227bf02a7E87c1e02B28C;
```
Comment (miguelmtzinf) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.sol:17]: You can use aave address book
https://github.com/bgd-labs/aave-address-book/blob/main/src/MiscEthereum.sol#L54

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AaveV3ArbitrumAssets} from 'aave-address-book/AaveV3Arbitrum.sol';
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+
+interface IGHOToken {
```
Comment (miguelmtzinf) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol:7]: Gho lib is part of the repo so I think it's better to directly import the interface

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+/**
+ * @title Increase CCIP Facilitator Capacity
+ * @author @karpatkey_TokenLogic
+ * - Snapshot: Direct-to-AIP
+ * - Discussion: https://governance.aave.com/t/arfc-gho-arb-increase-ccip-facilitator-capacity/18169
+ */
+contract AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707 is IProposalGenericExecutor {
+  address public constant FACILITATOR = 0xF168B83598516A532a85995b52504a2Fa058C068;
```
Comment (miguelmtzinf) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol:18]: you can use aave addresses book here too

**Comment by miguelmtzinf** [comment]
Changes to the bridge limit and the facilitator bucket capacity of CCIP on Arbitrum can result in users being locked. This occurs because the minting of GHO at the destination can revert if the bucket capacity is maxed out. More information about this issue can be found in the [governance discussion](https://governance.aave.com/t/arfc-gho-cross-chain-launch/17616).

This creates a poor UX, as users have no immediate solution other than to "wait and retry"—waiting for a bucket capacity increase to retry and get their tokens minted. The [CCIP explorer](https://ccip.chain.link/) provides a user-friendly UI for transaction retries.

Lock-ups can occur with both bucket capacity increases and decreases. However, bucket capacity decreases are more problematic because users will remain locked until the bucket capacity is raised again, which might not happen soon. In the case of bucket increases, users can get temporarily locked until the bucket capacity change is applied at the destination.

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol (hunk 2 above)
Comment (efecarranza) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol:7]: done

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity_20240707.s.sol [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+/**
+ * @dev Create Proposal
+ * command: make deploy-ledger contract=src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity_20240707.s.sol:CreateProposal chain=mainnet
+ */
+contract CreateProposal is EthereumScript {
+  function run() external {
+    // create payloads
+    PayloadsControllerUtils.Payload[] memory payloads = new PayloadsControllerUtils.Payload[](1);
```
Comment (brotherlymite) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity_20240707.s.sol:38]: @efecarranza Missing payload for arbitrum for creating the proposal

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+This publication proposes increasing the CCIP Facilitator Capacity on Arbitrum from 1M units to 2.5M units via the Direct-to-AIP process.
+
+## Motivation
+
+GHO was successfully launched on Arbitrum and the CCIP GHO Facilitator Bucket Capacity is 65% utilized. With $3.41M of liquidity and a solid peg during launch, there is sufficient depth to facilitate increasing the CCIP GHO Facilitator Bucket Capacity from 1M to 2.5M in line with the initial risk parameters proposed prior to launch 3.
+
+## Specification
+
+This proposal will increase the CCIP GHO Facilitator Bucket Capacity from 1M units to 2.5M units via the Direct-to-AIP process.
```
Comment (brotherlymite) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md:18]: I would also mention on the specification that we increase the bridgeLimit on ethereum as well

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.t.sol [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ * command: FOUNDRY_PROFILE=mainnet forge test --match-path=src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.t.sol -vv
+ */
+contract AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707_Test is ProtocolV3TestBase {
+  AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707 internal proposal;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 20257789);
+    proposal = new AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707();
+  }
+
```
Comment (brotherlymite) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.t.sol:26]: Similar to this [test](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240528_Multi_GHOCrossChainLaunch/AaveV3E2E_GHOCrossChainLaunch_20240528.t.sol#L241), I would add an e2e bridge test where before proposal execution we expect the the test to fail when facilitator is full, and after the limit increase we can bridge fine.

**Comment by brotherlymite** [comment]
Does it make sense that we pause robot execution for both the payloads and execute it manually on both networks together to prevent any gap. As the general case is the payload on ethereum gets executed 5-10mins before arbitrum so it could be the case the `bridgeLimit` on ethereum is increased before the facilitator limit on arbitrum, and if the facilitator was full before the user would need to retry on arbitrum when payload on arbitrum is executed. Or it its not a big issue and ask the user to retry, we can keep it the way it is.
Opinion @miguelmtzinf @efecarranza?

**Comment by miguelmtzinf** [comment]
Given that it is a bucket capacity increase and destination is not maxed out (still 600k GHO to hit the limit), it's unlikely to have user lock ups. However, we can disable Robot in this case so steps can be executed in the optimal order to do not diminish UX.

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md (hunk 5 above)
Comment (efecarranza) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md:18]: good point, added.

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity_20240707.s.sol (hunk 4 above)
Comment (efecarranza) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity_20240707.s.sol:38]: thanks

**Comment by efecarranza** [comment]
@brotherlymite @miguelmtzinf i added the e2e test and general modifications.
however, now capacity is at the limit so we might need to pause the bot for this execution.

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol (hunk 3 above)
Comment (brotherlymite) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol:15]: @efecarranza you can add it from here: https://github.com/bgd-labs/aave-address-book/blob/main/src/MiscArbitrum.sol#L31

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+The equivalent bridgeLimit is also being increased on Ethereum Mainnet to 2.5M units.
+
+## References
+
+- Implementation: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.sol)
+- Tests: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity_20240707.t.sol)
+- [Snapshot](Direct-to-AIP)
```
Comment (brotherlymite) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md:25]:
```suggestion
- Snapshot: Direct-to-AIP
```

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+snapshot: "Direct-to-AIP"
+---
+
+## Simple Summary
+
+This publication proposes increasing the CCIP Facilitator Capacity on Arbitrum from 1M units to 2.5M units via the Direct-to-AIP process.
+
+## Motivation
+
+GHO was successfully launched on Arbitrum and the CCIP GHO Facilitator Bucket Capacity is 65% utilized. With $3.41M of liquidity and a solid peg during launch, there is sufficient depth to facilitate increasing the CCIP GHO Facilitator Bucket Capacity from 1M to 2.5M in line with the initial risk parameters proposed prior to launch.
```
Comment (brotherlymite) [inline src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md:14]: @efecarranza Bucket Capacity currently is more than 65% currently no?

Comment (Matt-Graham-TL) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/IncreaseCCIPFacilitatorCapacity.md:14]: Currently, it is at 100% atm. It was 65% at time we wrote the hackmd though. Good catch. 
Ref. Bottom of this dashboard. https://aave.tokenlogic.xyz/gho

File: src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol (hunk 3 above)
Comment (efecarranza) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol:15]: done

Comment (brotherlymite) [reply src/20240707_AaveV3Ethereum_IncreaseCCIPFacilitatorCapacity/AaveV3Arbitrum_IncreaseCCIPFacilitatorCapacity_20240707.sol]: can't see

**Review by pavelvm5 — APPROVED** [review]

**Review by brotherlymite — APPROVED** [review]

**Pull request merged**